src/main/java/io/lumigo/models/BaseSpan.java

```diff
@@ -0,0 +1,3 @@
+package io.lumigo.models;
+
+public interface BaseSpan {}
```

Comment:
don't you want to move common fields to this base span?

started, ended, id, etc.